src/_pytest/fixtures.py

```diff
@@ -1765,6 +1767,80 @@ def _register_fixture(
         if autouse:
             self._nodeid_autousenames.setdefault(nodeid or "", []).append(name)
 
+    def _find_wrapped_fixture_def(
+        self, obj: object
+    ) -> FixtureFunctionDefinition | None:
+        """Walk through wrapper chain to find a FixtureFunctionDefinition.
+
+        Returns the FixtureFunctionDefinition if found in the wrapper chain,
+        None otherwise. Handles loops and special objects safely.
+        """
+        from _pytest.compat import safe_getattr
+
+        # Skip mock objects (they have _mock_name attribute)
+        if safe_getattr(obj, "_mock_name", None) is not None:
+            return None
+
+        current = obj
+        seen = set()  # Track object IDs to detect loops
+        max_depth = 100  # Prevent infinite loops even if ID tracking fails
+
+        for _ in range(max_depth):
+            if current is None:
+                break
+
+            # Check for wrapper loops by object identity
+            current_id = id(current)
+            if current_id in seen:
+                return None
+            seen.add(current_id)
+
+            # Check if current is a FixtureFunctionDefinition
+            # Use try/except to handle objects with problematic __class__ properties
+            try:
+                if isinstance(current, FixtureFunctionDefinition):
+                    return current
+            except Exception:
+                # Can't check isinstance - probably a proxy object
+                return None
+
+            # Try to get the next wrapped object using safe_getattr to handle
+            # "evil objects" that raise on attribute access (see issue #214)
+            wrapped = safe_getattr(current, "__wrapped__", None)
+            if wrapped is None:
+                break
+
+            current = wrapped
```

Comment:
```suggestion
            current = safe_getattr(current, "__wrapped__", None)
```
redundant with start-of-loop logic